src/webpack-angular-types-plugin/type-extraction/type-extraction.ts

```diff
@@ -15,13 +24,70 @@ function getterOrSetterInputExists(
     return !!entity && entity.kind === "input";
 }
 
+/**
+ * Creates mappings from generic class arguments of a subClass to the generic
+ * type parameters of the parent class. References are stored in weakMaps using
+ * ts-morph symbols (which are actually classes) to be able to resolve encountered
+ * types in a class by reference.
+ *
+ * Example, given the following two classes:
+ *  class ChildComponent extends ParentComponent<string, number>{}
+ *  class ParentComponent<S, T> {}
+ *
+ * The resulting mapping would look something like this:
+ * WeakMap {
+ *     <Symbol of ParentComponent> => WeakMap {
+ *          <Symbol of S> => string
+ *          <Symbol of T> => number
+ *     }
+ *}
+ *
+ * This information can later be used to map properties in base-classes to the
+ * actual type that is available when creating type-docs for a subClass.
+ */
+function extractGenericTypesFromClass(
+    classDeclarations: ClassDeclaration[]
+): GenericTypeMapping {
+    const result = new WeakMap<TsMorphSymbol, Type>();
+    for (const classDeclaration of classDeclarations) {
+        const targetTypeArguments = classDeclaration
+            .getExtends()
+            ?.getTypeArguments()
+            .map((a) => a.getType());
+        if (targetTypeArguments && targetTypeArguments.length > 0) {
+            const baseClass = classDeclaration.getBaseClass();
+            if (baseClass) {
+                const sourceTypeArguments = baseClass
+                    .getTypeParameters()
+                    .map((p) => p.getType());
+                if (
+                    sourceTypeArguments &&
+                    sourceTypeArguments.length === targetTypeArguments.length
+                ) {
+                    for (let i = 0; i < sourceTypeArguments.length; i++) {
+                        const mappedType = tryToReplaceTypeByGenericType(
+                            targetTypeArguments[i],
+                            result
+                        );
+                        result.set(
+                            sourceTypeArguments[i].getSymbolOrThrow(),
+                            mappedType
+                        );
+                    }
+                }
+            }
+        }
+    }
+    return result;
+}
```

Comment:
I believe this can be easier to read. Something like this, what do you think?

```typescript
function extractGenericTypesFromClass(
  classDeclarations: ClassDeclaration[]
): GenericTypeMapping {
  const result = new WeakMap<TsMorphSymbol, Type>();
  for (const classDeclaration of classDeclarations) {
    const baseClass = classDeclaration.getBaseClass();
    const targetTypeArguments = classDeclaration
      .getExtends()
      ?.getTypeArguments()
      .map((a) => a.getType());

    if (!targetTypeArguments?.length || !baseClass) {
      continue;
    }

    const sourceTypeArguments = baseClass
      .getTypeParameters()
      .map((p) => p.getType());
    
    if (
      !sourceTypeArguments ||
      sourceTypeArguments.length !== targetTypeArguments.length
    ) {
      continue;
    }
    
    for (let i = 0; i < sourceTypeArguments.length; i++) {
      const mappedType = tryToReplaceTypeByGenericType(
        targetTypeArguments[i],
        result
      );
      result.set(sourceTypeArguments[i].getSymbolOrThrow(), mappedType);
    }
  }
  return result;
}
```

Reply:
I significantly simplified this method by using a mapping helper i already wrote for generic mapping during type-crawling in the `type-details` module (just forgot to add it for class generic too). Should be much easier to read now.